File: docs/user/en/use/api/example/query-example.java
final EvitaResponse<SealedEntity> response = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					and(
						hierarchyWithin(
							"categories",
							attributeEquals("url", "/en/macbooks")
						),
						entityLocaleEquals(Locale.ENGLISH),
						priceInPriceLists("basic"),
						priceInCurrency(Currency.getInstance("USD"))
					)
				),
				require(
					page(2, 24),
					entityFetch(
						attributeContentAll(),
						priceContentRespectingFilter()
					),
					facetSummary()
				)
			)
		);
	}
);
PaginatedList<SealedEntity> paginatedList = (PaginatedList<SealedEntity>) response.getRecordPage();
List<SealedEntity> entities = paginatedList.getData();
int pageNumber = paginatedList.getPageNumber();
int pageSize = paginatedList.getPageSize();
int totalRecordCount = paginatedList.getTotalRecordCount();
FacetSummary facetSummary = response.getExtraResult(FacetSummary.class);

File: documentation/user/en/use/api/example/query-example.cs
EvitaResponse<ISealedEntity> response = evita.QueryCatalog(
	"evita",
	session => {
		return session.QuerySealedEntity(
			Query(
				Collection("Product"),
				FilterBy(
					And(
						HierarchyWithin(
							"categories",
							AttributeEquals("url", "/en/macbooks")
						),
						EntityLocaleEquals(new CultureInfo("en")),
						PriceInPriceLists("basic"),
						PriceInCurrency(new Currency("USD"))
					)
				),
				Require(
					Page(2, 24),
					EntityFetch(
						AttributeContentAll(),
						PriceContentRespectingFilter()
					),
					FacetSummary()
				)
			)
		);
	}
);

PaginatedList<ISealedEntity> paginatedList = (PaginatedList<ISealedEntity>) response.RecordPage;
IList<ISealedEntity> entities = paginatedList.Data;
int pageNumber = paginatedList.PageNumber;
int pageSize = paginatedList.PageSize;
int totalRecordCount = paginatedList.TotalRecordCount;
FacetSummary facetSummary = response.GetExtraResult<FacetSummary>();
